src/main/java/Group15/WorkoutAlgorithm.java

```diff
@@ -0,0 +1,51 @@
+package Group15;
+
+import Group15.Api.ApiUtils;
+import Group15.Api.Exercise;
+
+import java.util.List;
+
+public class WorkoutAlgorithm {
+    public Workout createWorkoutFromExercises(String bodyPart, String equipment, int timeInMinutes) {
+        List<Exercise> exercises = ApiUtils.getAllExercises();
+
+        int timeInMilliseconds = timeInMinutes * 60000;
+        Workout workout = new Workout();
+
+        System.out.println("Workout");
+
+        if (exercises != null) {
+            for (Exercise exercise : exercises) {
+                if (exercise.bodyPart.equals(bodyPart)) {
+                    boolean equipmentMatch = equipment.equals("All") || exercise.equipment.equals(equipment);
```

Comment:
The way you are looping through your elements will give you the same workout given the same parameters